packages/react/src/Button/Button.module.css

```diff
@@ -74,7 +74,7 @@
   background: var(--brand-Button-background-overlay);
 }
 
-.Button--primary:not(.Button--disabled):hover {
+.Button--primary:not(.Button--disabled[disabled]):hover {
```

Comment:
Quite odd that `.Button--disabled` isn't doing its job here. Do you know why?

Does this have the same effect? Wonder if it's worth just dropping the `.Button--disabled` class if attribute approach works just as well.

```jsx
.Button--primary:not(.Button[disabled]):hover {
```

Also, just an FYI, that in the example you shared the secondary button retains its focus on press. 


https://user-images.githubusercontent.com/13340707/225414637-e8270d07-1abe-41d8-b5fd-845e61cff4fa.mov



Can we make the effect consistent between all variants?

Reply:
Ah good callout, and thanks for the suggestion! I added the same rules for the secondary button variant. I also slimmed down the selector as recommended.

As for `.Button--disabled` not working as expected, I believe it's due to one selector being more specific than the other. It seems if I add `[disabled]` in one place, I need to make sure the other styles that'll apply in those scenarios also have the same `[disabled]` or `[aria-disabled]` selectors, as to keep the same priority in which styles overwrite others 🤔.

I also added `[disabled]` in some cases, because I only wanted those styles to apply when `[disabled]` and not `[aria-disabled]`.

Reply:
Added it to the `subtle` variant! 

I think the original button styles had the same focus/button press interactions, where the hover style took priority over the focus style, but I'm not entirely sure.